src/main/java/nextstep/qna/service/QnAService.java

```diff
@@ -26,24 +27,11 @@ public class QnAService {
     @Transactional
     public void deleteQuestion(NsUser loginUser, long questionId) throws CannotDeleteException {
         Question question = questionRepository.findById(questionId).orElseThrow(NotFoundException::new);
-        if (!question.isOwner(loginUser)) {
-            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
-        }
-
-        List<Answer> answers = question.getAnswers();
-        for (Answer answer : answers) {
-            if (!answer.isOwner(loginUser)) {
-                throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
-            }
-        }
-
-        List<DeleteHistory> deleteHistories = new ArrayList<>();
-        question.setDeleted(true);
-        deleteHistories.add(new DeleteHistory(ContentType.QUESTION, questionId, question.getWriter(), LocalDateTime.now()));
-        for (Answer answer : answers) {
-            answer.setDeleted(true);
-            deleteHistories.add(new DeleteHistory(ContentType.ANSWER, answer.getId(), answer.getWriter(), LocalDateTime.now()));
-        }
+
+        question.delete(loginUser);
+
+        List<DeleteHistory> deleteHistories = question.toDeleteHistories();
```

Comment:
서비스가 굉장히 얇아졌군요 👏 
좋습니다 👍 

```suggestion
        List<DeleteHistory> deleteHistories = question.delete(loginUser);
```

`질문을 삭제하면 삭제 이력이 생성된다.` 와 같은 책임은 어떨까요?